app/components/scheduled_post_indicator/index.ts

```diff
@@ -2,23 +2,46 @@
 // See LICENSE.txt for license information.
 
 import {withDatabase, withObservables} from '@nozbe/watermelondb/react';
-import {map} from 'rxjs/operators';
+import {map, switchMap} from 'rxjs/operators';
 
 import {getDisplayNamePreferenceAsBool} from '@helpers/api/preference';
 import {queryDisplayNamePreferences} from '@queries/servers/preference';
+import {observeScheduledPostCountForChannel, observeScheduledPostCountForDMsAndGMs, observeScheduledPostCountForThread} from '@queries/servers/scheduled_post';
+import {observeCurrentChannelId, observeCurrentTeamId} from '@queries/servers/system';
 import {observeCurrentUser} from '@queries/servers/user';
 
 import {ScheduledPostIndicator} from './scheduled_post_indicator';
 
-const enhance = withObservables([], ({database}) => {
+import type {WithDatabaseArgs} from '@typings/database/database';
+
+type Props = WithDatabaseArgs & {
+    channelId?: string;
+    channelType?: ChannelType;
+    isCRTEnabled: boolean;
+    rootId?: string;
+}
+
+const enhance = withObservables(['channelId', 'channelType', 'isCRTEnabled', 'rootId'], ({database, channelId, channelType, isCRTEnabled, rootId}: Props) => {
     const currentUser = observeCurrentUser(database);
+    const currentTeamId = observeCurrentTeamId(database);
+    const currentChannelId = observeCurrentChannelId(database);
     const preferences = queryDisplayNamePreferences(database).
         observeWithColumns(['value']);
     const isMilitaryTime = preferences.pipe(map((prefs) => getDisplayNamePreferenceAsBool(prefs, 'use_military_time')));
 
+    let scheduledPostCount;
+    if ((channelType === 'D' || channelType === 'G') && channelId) {
+        scheduledPostCount = observeScheduledPostCountForDMsAndGMs(database, channelId, isCRTEnabled);
+    } else if (channelType === 'O' && channelId) {
+        scheduledPostCount = currentTeamId.pipe(switchMap((teamId) => observeScheduledPostCountForChannel(database, teamId, channelId, isCRTEnabled)));
+    } else if (rootId) {
```

Comment:
If its a thread in open channel, the code will go the the above else if, not this else, right? That looks wrong.

Reply:
I have reverted the code, Now we will be checking firstly if rootId is present or not, if present that means we have to extract the count of the thread. If not then extract the count of the channel, DMs and GMs.